core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/BaseResourceSearchRequest.java

```diff
@@ -98,6 +103,21 @@ protected final Expression prepareQuery(RepositoryContext context) {
 				.build());
 		}
 
+		if (containsKey(OptionKey.SETTINGS)) {
+			final Collection<String> properties = getCollection(OptionKey.SETTINGS, String.class);
+			properties.forEach( property -> {
+				if (property.contains("#")) {
+					final String propertyName = property.split("\\#")[0];
+					final String propertyValue = property.substring(propertyName.length() + 1, property.length());
```

Comment:
If I specify `key#` without value this will throw an Exception, please handle that and create a test case for it.